app/src/test/java/org/mozilla/vrbrowser/GleanMetricsServiceTest.kt

```diff
@@ -0,0 +1,150 @@
+package org.mozilla.vrbrowser
+
+import androidx.test.core.app.ApplicationProvider
+import mozilla.components.concept.sync.DeviceType
+import mozilla.components.service.glean.testing.GleanTestRule
+import org.junit.Assert.*
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.mozilla.vrbrowser.GleanMetrics.Distribution
+import org.mozilla.vrbrowser.GleanMetrics.FirefoxAccount
+import org.mozilla.vrbrowser.GleanMetrics.Tabs
+import org.mozilla.vrbrowser.GleanMetrics.Url
+import org.mozilla.vrbrowser.telemetry.GleanMetricsService
+import org.robolectric.RobolectricTestRunner
+import org.robolectric.annotation.Config
+
+
+@RunWith(RobolectricTestRunner::class)
+@Config(manifest = Config.NONE)
+class GleanMetricsServiceTest {
+
+    @get:Rule
+    val gleanRule = GleanTestRule(ApplicationProvider.getApplicationContext())
+
+    @Test
+    fun testMetricsGathering() {
+        testURLTelemetry()
+        testDistributionChannelTelemetry()
+        testFxAAccountTelemetry()
+        testFxABookmarkTelemetry()
+        testFxAHistoryTelemetry()
+        testFxATabTelemetry()
+        testTabTelemetry()
+    }
```

Comment:
I'd recommend, if you agree, to make all the test cases isolated: instead of calling them within this function marked as `@Test`, remove this and prefix all other functions with `@Test`.

This will make sure that all tests run separately and don't rely on the state from a previous test. If something breaks, this make it easier to track down problems. It also makes it simpler to re-execute tests one by one and to fix intermittent failures, in case they pop up.